### PR TITLE
Improve RLP API

### PR DESCRIPTION
**Pull request by turbolent**
## Description

- Refactor the RLP functions into the built-in contract `RLP`
- Refactor the `stdlib` package, by splitting the code into separate files
- Add a status note to the documentation
- Remove the argument labels from the functions, to make the API consistent with e.g. the BLS functions
- Simplify and parallelize tests. Also, move them to the runtime package, just like where the BLS tests are
- Add location/position information to the errors to improve error messages
- Add checker tests for valid uses

______

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/crypto_test.go [hunk 1]
```diff
@@ -708,9 +708,9 @@ func TestTraversingMerkleProof(t *testing.T) {
                 }
             }
 
-            let encodedChildren = DecodeRLPList(input: encodedNode)
+            let encodedChildren = RLP.decodeList(encodedNode)
```
Comment (ramtinms) [inline runtime/crypto_test.go:711]: do I need to call import RLP here ?

**Review by ramtinms — APPROVED** [review]
I only checked RLP related changes, and looked good to me.

File: runtime/stdlib/assert.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	),
+	RequiredArgumentCount: sema.RequiredArgumentCount(1),
+}
+
+var AssertFunction = NewStandardLibraryFunction(
+	"assert",
+	assertFunctionType,
+	assertFunctionDocString,
+	func(invocation interpreter.Invocation) interpreter.Value {
+		result := invocation.Arguments[0].(interpreter.BoolValue)
```
Comment (dsainati1) [inline runtime/stdlib/assert.go:58]: Check this cast

File: runtime/stdlib/assert.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		if !result {
+			var message string
+			if len(invocation.Arguments) > 1 {
+				message = invocation.Arguments[1].(*interpreter.StringValue).Str
```
Comment (dsainati1) [inline runtime/stdlib/assert.go:62]: Check this cast

File: runtime/stdlib/hashalgorithm.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		inter,
+		sema.HashAlgorithmType,
+		interpreter.UInt8Value(rawValue),
+		hashAlgorithmFunctions,
+	)
+}
+
+var hashAlgorithmHashFunction = interpreter.NewHostFunctionValue(
+	func(invocation interpreter.Invocation) interpreter.Value {
+		dataValue := invocation.Arguments[0].(*interpreter.ArrayValue)
```
Comment (dsainati1) [inline runtime/stdlib/hashalgorithm.go:43]: Check this cast

File: runtime/stdlib/hashalgorithm.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+			hashAlgoValue,
+		)
+	},
+	sema.HashAlgorithmTypeHashFunctionType,
+)
+
+var hashAlgorithmHashWithTagFunction = interpreter.NewHostFunctionValue(
+	func(invocation interpreter.Invocation) interpreter.Value {
+		dataValue := invocation.Arguments[0].(*interpreter.ArrayValue)
+		tagValue := invocation.Arguments[1].(*interpreter.StringValue)
```
Comment (dsainati1) [inline runtime/stdlib/hashalgorithm.go:70]: Check these casts

File: runtime/stdlib/panic.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+				TypeAnnotation: sema.NewTypeAnnotation(sema.StringType),
+			},
+		},
+		ReturnTypeAnnotation: sema.NewTypeAnnotation(
+			sema.NeverType,
+		),
+	},
+	panicFunctionDocString,
+	func(invocation interpreter.Invocation) interpreter.Value {
+		message := invocation.Arguments[0].(*interpreter.StringValue)
```
Comment (dsainati1) [inline runtime/stdlib/panic.go:57]: Check this cast

File: runtime/stdlib/publickey.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	ReturnTypeAnnotation: sema.NewTypeAnnotation(sema.PublicKeyType),
+}
+
+var publicKeyConstructor = NewStandardLibraryFunction(
+	sema.PublicKeyTypeName,
+	publicKeyConstructorFunctionType,
+	publicKeyConstructorFunctionDocString,
+	func(invocation interpreter.Invocation) interpreter.Value {
+		publicKey := invocation.Arguments[0].(*interpreter.ArrayValue)
+		signAlgo := invocation.Arguments[1].(*interpreter.CompositeValue)
```
Comment (dsainati1) [inline runtime/stdlib/publickey.go:50]: Check these casts

File: runtime/stdlib/assert.go (hunk 2 above)
Comment (turbolent) [reply runtime/stdlib/assert.go:58]: This is just moving code, but I added them separately in #1422. Will also add them here I guess

Comment (turbolent) [reply runtime/stdlib/assert.go]: Added in 8790562c756b4bf654d20b6da95cfe5ec8dda56c

**Review by SupunS — APPROVED** [review]
Nice idea to split the builtin stuff into separated files 👌

**Pull request merged**